### DISCUSSION TITLE
Jupyter Notebook 隱藏技能解鎖

### DISCUSSION THREAD

**Discussion by Bill671** [1101-GS3073-C/Python-Community-GS3073-C#270] [Ideas]
大家都會用Jupyter notebook吧，但Jupyter notebook 有個缺點，那就是每次看變數都要用print才看的到，這相對於spyder有變數查看器來說非常不方便，雖然Jupyter的優點就是逐步執行，可以把程式碼切分一小塊一小塊，最後再組在一起，但當我今天匯入的是一筆大型資料，需要每個步驟都把變數印出來便是他的硬傷。
所幸，你想的到的問題Jupyter 開發者也想到了，為了因應功能上的不足，
Jupyter有個隱藏技能 jupyter nbextensions，先說推薦打勾的項目跟原因，再說怎麼安裝。
首先必定要打勾的是 Table of Contents，如果你在撰寫Jupyter時的markdown 有按照層次 h1-hn撰寫，那麼TOC就會自動幫你生成目錄，讓你在撰寫代碼時能快速跳到頁面中的各個位置，第二個是variable inspector，它可以看到你變數長怎樣占多少空間，雖說仍然不比Spyder好用，但已經可以讓你看到基本的像是shape跟一些values。

至於怎麼安裝，首先打開window輸入anaconda powershell prompt，開啟後輸入`pip install jupyter_contrib_nbextensions`
輸入完成後啟動Jupyter，會看到旁邊多了一個nbextension，接著就可以勾選你想要的擴展元件。

[詳細說明請點我](https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/install.html)

**Comment by enoch-lightning** [comment]
你說的終端機，應該就是命令提示字元。但我在命令提示字元打anaconda prompt，並按下enter後，怎麼會出現`'anaconda' 不是內部或外部命令、可執行的程式或批次檔`？anaconda prompt應該是安裝anaconda時順便會安裝的東西，因此我有安裝anaconda prompt。
![image](https://user-images.githubusercontent.com/86505454/156551945-9a14be64-5b7c-40d8-8ac1-71460cbc3a05.png)

![image](https://user-images.githubusercontent.com/86505454/156551888-83af2042-71d5-4f61-9e11-0e2c69a33882.png)

**Reply by fatsoap** [reply]
打開 anaconda prompt 然後輸入 pip .... 即可。
從終端機打開應該是口誤

**Reply by enoch-lightning** [reply]
原來如此，我來試試看。我看了[非官方jupyter插件](https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/install.html)，也就是 @Bill671 在上面的[詳細說明請點我](https://jupyter-contrib-nbextensions.readthedocs.io/en/latest/install.html)，這個網站上面的說明是說用安裝原生python時，會順便安裝的pip這個package manager(我不知道pip是否也是一個package)安裝`jupyter_contrib_nbextensions`即可。

**Reply by Bill671** [reply]
> 打開 anaconda prompt 然後輸入 pip .... 即可。 從終端機打開應該是口誤

對，口誤，感謝fatsoap 大神補充

**Reply by Bill671** [reply]
但是是要在 anaconda powershell 打指令安裝，原因是因為window數位簽章的問題

@enoch-lightning 按照官網步驟執行應該就可以多了那個功能

**Comment by lx3211** [comment]
vscode裡面也可以使用jupyter插件，也可以隨時追蹤變數喔

**Comment by NcuJerryKao** [comment]
感謝啦 這樣就不用再打完 Jupyter 而後切spyder了

**Comment by limitx0** [comment]
用 Google Colab 的話左邊也有變數欄位可以追蹤變數
